server/rpc.go

```diff
@@ -35,6 +36,11 @@ const GRPCTimeout = 8 * time.Second
 
 var authTokenValidPeriod = 30 * time.Minute
 
+// using authTokenValidPeriod minus 1 minute as cache duration. With a 30m duration, worst case we could end up with cache items that expire after 30 minutes
```

Comment:
Why set the default cleanup interval to 29 minutes here? A 29 minute cleanup interval doesn't seem like it would result in much of a difference vs. a 30 minute cleanup interval.

Thinking back to my previous comment, it also seems like the cleanup interval value here may end up not being that important because if a cached response is stale, then that should mean the auth token used by a B has expired which would require the B to refresh the auth token triggering a webhook call that would overwrite the expired cached response anyway.

Reply:
My bad , I misunderstood your earlier comment